packages/core/admin/admin/src/pages/Home/components/Widget.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+
+import { Box, Flex, Loader, Typography } from '@strapi/design-system';
+import { PuzzlePiece, WarningCircle } from '@strapi/icons';
+import { EmptyDocuments } from '@strapi/icons/symbols';
+import { type MessageDescriptor, useIntl } from 'react-intl';
+
+interface RootProps {
+  title: MessageDescriptor;
```

Comment:
What's your reasoning for requiring the translation object vs just a string? The argument I see for the string is the Widget consumer decides whether or not the title should be translated.

Reply:
I'm thinking that the future API will likely look something like this:

```js
bootstrap() {
  app.registerWidget({ title, icon, component })
}
```

and in that context, from a lifecycle function, I'm not sure how to access the react-intl utils. Custom fields are a quite similar and we pass the translation object for them

Reply:
What do you need title and icon for when registering a widget? Are they used elsewhere besides the component itself?

Reply:
not for the mvp, but we will need them in the future for a "configure the view" style feature